armada/api/tiller_controller.py

```diff
@@ -13,53 +13,74 @@
 # limitations under the License.
 #
 
-from armada.conf import default
-
 import json
-from falcon import HTTP_200
-
-from oslo_config import cfg
-from oslo_log import log as logging
-
-# Required Oslo configuration setup
-default.register_opts()
+from falcon import HTTP_200, HTTP_404
 
-from armada.handlers.tiller import Tiller as tillerHandler
+from armada import api
+from armada.handlers.tiller import Tiller
 
-LOG = logging.getLogger(__name__)
-CONF = cfg.CONF
-DOMAIN = "armada"
-
-logging.setup(CONF, DOMAIN)
-
-class Status(object):
+class Status(api.BaseResource):
     def on_get(self, req, resp):
         '''
         get tiller status
         '''
-        message = "Tiller Server is {}"
-        if tillerHandler().tiller_status():
-            resp.data = json.dumps({'message': message.format('Active')})
-            LOG.info('Tiller Server is Active.')
-        else:
-            resp.data = json.dumps({'message': message.format('Not Present')})
-            LOG.info('Tiller Server is Not Present.')
+        try:
+            policy_action = 'chart_installer:query_status'
+            ctx = req.context
+
+            if not self.check_policy(policy_action, ctx):
+                self.access_denied(req, resp, policy_action)
+                return
 
-        resp.content_type = 'application/json'
-        resp.status = HTTP_200
+            message = {
+                'tiller': Tiller().tiller_status()
+            }
 
-class Release(object):
+            if message.get('tiller'):
+                resp.data = json.dumps(message)
+                resp.status = HTTP_200
+            else:
+                resp.data = json.dumps(message)
+                resp.status = HTTP_404
+
+            resp.content_type = 'application/json'
+        except Exception:
```

Comment:
A 404 in this case would be a server error rather than a client error. Would a 503 make sense here? (services temporarily unavailable)

Or a 2xx for a successful request since the Tiller status is already listed in the resp message